### PR TITLE
HPCC-25315 Improve superfile transaction lock tracing

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by mckellyln — DISMISSED** [review]
Looks fine.
Approved.

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -1174,7 +1176,13 @@ class CDFAction: public CInterface
                 if (SDSExcpt_LockTimeout != e->errorCode())
                     throw;
                 e->Release();
-                PROGLOG("CDFAction lock timed out on %s",lockedFiles.item(i).queryLogicalName());
+                PROGLOG("CDFAction[%s] lock timed out on %s", tracing.str(), lockedFiles.item(i).queryLogicalName());
+
+                /* Can be v. useful to know what call stack is if stuck..
+                 * Trace after 30 timesouts (each timeout period ~60s)
```
Comment (richardkchapman) [inline dali/base/dadfs.cpp:1182]: typo timesouts

**Review by richardkchapman — COMMENTED** [review]
Couple of minor quibbles

File: dali/base/dadfs.cpp [hunk 2]
```diff
@@ -1150,11 +1150,13 @@ protected: friend class CDistributedFile;
...
-    unsigned locked;
+    unsigned locked = 0;
+    unsigned timeoutCount = 0;
 protected:
-    IDistributedFileTransactionExt *transaction;
+    IDistributedFileTransactionExt *transaction = nullptr;
     IArrayOf<IDistributedFile> lockedFiles;
-    DFTransactionState state;
+    DFTransactionState state = TAS_NONE;
+    StringBuffer tracing;
```
Comment (richardkchapman) [inline dali/base/dadfs.cpp:1159]: Should really be a const StringAttr (though would need to add a printf-style constructor to StringAttr...)

Comment (jakesmith) [reply dali/base/dadfs.cpp:1159]: I'd prefer to leave that for another time/PR.
Would probably want to share some of the code that StringBuffer currently employs to check that the expanded printf string is not too big, and then to truncate after.

File: dali/base/dadfs.cpp (hunk 1 above)
Comment (jakesmith) [reply dali/base/dadfs.cpp]: fixed

**Comment by jakesmith** [comment]
@richardkchapman - fixed typo' in comment, squashed in and rebased to current master.

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**